roles/install_yamls/tasks/zuul_set_operators_repo.yml

```diff
@@ -0,0 +1,36 @@
+---
+# Copyright Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+# When using CI (Zuul) to deploy OpenStack control plane with install_yamls,
+#  it may be needed to set operator's repo variable to properly clone PR's
+#  code, instead of getting latest promoted content. This task search for all
+#  modified operators in zuul.items[] and set install_yaml variables.
+- name: Create variables with local repos based on Zuul items
+  when:
+    - zuul is defined
+    - "'operator' in zuul_item.project.short_name"
+  vars:
+    _repo_operator_name: "{{ zuul_item.project.short_name | regex_search('(?:openstack-)?(.*)-operator', '\\1') | first }}"
+    _repo_operator_info:
+      - key: "{{ _repo_operator_name | upper }}_REPO"
+        value: "{{ ansible_user_dir }}/{{ zuul_item.project.src_dir }}"
+      - key: "{{ _repo_operator_name | upper }}_BRANCH"
+        value: ""
+  ansible.builtin.set_fact:
+    cifmw_install_yamls_operators_repo: "{{ cifmw_install_yamls_operators_repo | default({}) | combine(_repo_operator_info | items2dict) }}"
```

Comment:
I'm pretty sure we would like to have this variable here: https://github.com/openstack-k8s-operators/ci-framework/blob/main/roles/install_yamls/tasks/main.yml#L123-L134

Reply:
This play is not integrated with the main install_yamls role, which is used to generate the install_yamls_makes role, so at that point, there is no "cifmw_install_yamls_operators_repo". This zuul_set_operators_repo.yml is being called at specific points, like edpm_prepare, edpm_deploy, to set up repo vars. My original thought is that they do different things (the main role and this play), and I didn't combine them. I am open to different ideas around that too. Tks

Reply:
Thid is now combined with cifmw_install_yamls_environment: https://github.com/openstack-k8s-operators/ci-framework/pull/1177/files#diff-9fb1a6e268d18db25f55afbd005cbeea219a6ca639e5296132cfcd9628af14d4R53